tests/test-awful-client.lua

```diff
@@ -77,6 +77,61 @@ end
 end
 }
 
+local original_count, c1, c2 = 0
+
+-- Check request::activate
+table.insert(steps, function()
+    c1, c2 = client.get()[1], client.get()[2]
+
+    -- This should still be the case
+    assert(client.focus == c1)
+
+    c2:emit_signal("request::activate", "i_said_so")
+
+    return true
+end)
+
+-- Check if writing a focus stealing filter works.
+table.insert(steps, function()
+    -- This should still be the case
+    assert(client.focus == c2)
+
+    original_count = #awful.ewmh.activate_filters
+
+    awful.ewmh.add_activate_filter(function(c)
+        if c == c1 then return false end
+    end)
+
+    c1:emit_signal("request::activate", "i_said_so")
+
+    return true
+end)
+
+table.insert(steps, function()
+    -- The request should have been denied
+    assert(client.focus == c2)
+
+    -- Test the remove function
+    awful.ewmh.remove_activate_filter(function() end)
+
+    awful.ewmh.add_activate_filter(awful.ewmh.activate_filters[1])
+
+    awful.ewmh.remove_activate_filter(awful.ewmh.activate_filters[1])
+
+    assert(original_count == #awful.ewmh.activate_filters)
+
+    c1:emit_signal("request::activate", "i_said_so")
+
+    return true
+end)
+
+
+table.insert(steps, function()
+    -- Now that the filter is gone, it should work again.
+
+    return client.focus == c1
+end)
```

Comment:
I think that you can merge all of this into a single step. While it is not always clear which operations are instantaneous and which aren't, setting the input focus works fine (except for some corner cases where the client then forwards the focus elsewhere).
Less steps = faster tests. :-)